distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -651,11 +652,17 @@ type Table
             resolved_aggregates = validated.valid_columns
             key_expressions = key_columns.map .expression
             new_ctx = this.context.set_groups key_expressions
-            new_columns = resolved_aggregates.map p->
+            results = resolved_aggregates.map p->
                 agg = p.second
                 new_name = p.first
-                Aggregate_Helper.make_aggregate_column this agg new_name
-            this.updated_context_and_columns new_ctx new_columns
+                Aggregate_Helper.make_aggregate_column this agg new_name . catch
+            partitioned = results.partition <| case _ of
```

Comment:
could we do `_.is_an Internal_Column` here?

Reply:
Good point, yes!